Monika After Story/game/zz_submods.rpy

```diff
@@ -0,0 +1,318 @@
+            self.version = version
+            self.dependencies = dependencies
+            self.settings_pane = settings_pane
+            self.version_updates = version_updates
+
+            #Now we add these to our maps
+            submod_map[name] = self
+
+            #NOTE: We check for things having updated later so all update scripts get called together
+            if name not in persistent._mas_submod_version_data:
+                persistent._mas_submod_version_data[name] = version
+
+        def getVersionNumberList(self):
+            """
+            Gets the version number as a list of integers
+
+            OUT:
+                List of integers representing the version number
+            """
+            return map(int, self.version.split('.'))
+
+        def hasUpdated(self):
+            """
+            Checks if this submod instance was updated (version number has incremented)
+
+            OUT:
+                boolean:
+                    - True if the version number has incremented from the persistent one
+                    - False otherwise
+            """
+            old_vers = persistent._mas_submod_version_data.get(self.name)
+
+            #If we don't have an old vers, we're installing for the first time and aren't updating at all
+            if not old_vers:
+                return False
+
+            try:
+                old_vers = map(int, old_vers.split('.'))
+
+            #Persist data was bad, we'll replace it with something safe and return False as we need not check more
+            except:
+                persistent._mas_submod_version_data[self.name] = Submod.FB_VERS_STR
+                return False
+
+            curr_vers = self.getVersionNumberList()
+
+            return Submod._checkVersions(curr_vers, old_vers, is_current=False)
+
+        def updateFrom(self, version):
+            """
+            Updates the submod, starting at the given start version
+
+            IN:
+                version - the version number in the parsed format ('author_name_v#_#_#')
+            """
+            while version in self.version_updates:
+                updateTo = self.version_updates[version]
+
+                # we should only call update labels that we have
+                if renpy.has_label(updateTo) and not renpy.seen_label(updateTo):
+                    renpy.call_in_new_context(updateTo, updateTo)
+                version = self.version_updates[version]
+
+        @staticmethod
+        def _checkUpdates():
+            """
+            Checks if submods have updated and sets the appropriate update scripts for them to run
+            """
+            #Iter thru all submods we've got stored
+            for submod in submod_map.itervalues():
+                #If it has updated, we need to call their update scripts and adjust the version data value
+                if submod.hasUpdated():
+                    submod.updateFrom(
+                        "{0}_{1}_v{2}".format(
```

Comment:
is that `v` supposed to be in front of the `{2}`?

Reply:
Yeah, that form is mainly to enforce a similar formatting to our update labels (`v0_10_x` for example) but also adding the author + name to it for differentiation and avoidance of overlap